oak_restricted_kernel/src/mm/page_tables.rs

```diff
@@ -279,9 +283,28 @@ impl CurrentRootPageTable {
     }
 
     /// Replaces the current pagetable with the parameter, returning the old
-    /// pagetable if there was one. Does not issue the instructions update the
-    /// page table on the CPU level.
-    pub fn replace(&mut self, new_pt: RootPageTable) -> Option<RootPageTable> {
+    /// pagetable if there was one. Updates the page table on the CPU level.
+    /// Safety: The new page tables must keep the identity mapping at -2GB (kernel space) intact.
+    pub unsafe fn replace(
+        &mut self,
+        pml4_frame: PhysFrame,
+        encrypted: MemoryEncryption,
+    ) -> Option<RootPageTable> {
+        // This validates any references that expect boot page tables to be valid!
+        // Safety: Caller must ensure that the new page tables are safe.
+        unsafe {
+            Efer::update(|flags| flags.insert(EferFlags::NO_EXECUTE_ENABLE));
```

Comment:
This should only be done once, you don't need to do it every time you update CR3.

Reply:
updated, thanks